File: ObjC/AoC2025/AoC2025/Utility/AOCSolution.m
//
//  AOCSolution.m
//  AoC2025
//
//  Created by Simon Biickert on 2023-01-27.
//

#import <Foundation/Foundation.h>
#import "AOCSolution.h"
#import "AOCDay.h"

@implementation AOCSolution

+ (NSArray<AOCSolution *> *)allSolutions {
	NSArray<AOCSolution *> *solutions = @[ [[AOCDay00 alloc] init],
										   [[AOCDay01 alloc] init],
//										   [[AOCDay02 alloc] init],
//										   [[AOCDay03 alloc] init],
//										   [[AOCDay04 alloc] init],
//										   [[AOCDay05 alloc] init],
//										   [[AOCDay06 alloc] init],
//										   [[AOCDay07 alloc] init],
//										   [[AOCDay08 alloc] init],
//										   [[AOCDay09 alloc] init],
//										   [[AOCDay10 alloc] init],
//										   [[AOCDay11 alloc] init],
//										   [[AOCDay12 alloc] init],
//										   [[AOCDay13 alloc] init],
//										   [[AOCDay14 alloc] init],
//										   [[AOCDay15 alloc] init],
//										   [[AOCDay16 alloc] init],
//										   [[AOCDay17 alloc] init],
//										   [[AOCDay18 alloc] init],
//										   [[AOCDay19 alloc] init],
//										   [[AOCDay20 alloc] init],
//										   [[AOCDay21 alloc] init],
//										   [[AOCDay22 alloc] init],
//										   [[AOCDay23 alloc] init],
//										   [[AOCDay24 alloc] init],
//										   [[AOCDay25 alloc] init]
	];
	return [[solutions reverseObjectEnumerator] allObjects];
}

- (AOCSolution *)initWithDay:(int)day name:(NSString *)name {
	self = [super init];
	_day = day;
	_name = name;
	self.emptyLinesIndicateMultipleInputs = YES;
	return self;
}

- (struct AOCResult)solveInputIndex:(int)index inFile:(NSString *)filename {
	NSLog(@"Day %02d: \"%@\" with input %@ [%d]", self.day, self.name, filename, index);
	struct AOCResult result;
	result.part1 = @"";
	result.part2 = @"";
	return result;
}

- (struct AOCResult)solveInput:(AOCInput *)input {
	return [self solveInputIndex:input.index inFile:input.filename];
}

@end

File: Swift/AoC2025/MyTool/Libraries/AoCSolution.swift
//
//  AoCSolution.swift
//  AoC 2025
//
//  Created by Simon Biickert on 2023-11-27.
//

import Foundation

struct AoCResult {
	let part1: String?
	let part2: String?
	
	var description: String {
		"Part 1: \(part1 ?? "")\nPart 2: \(part2 ?? "")"
	}
	var debugDescription: String {
		return description
	}
}

class AoCSolution {
	public static var solutions: [AoCSolution] {
		get {
			return [Day00()
			].reversed()
		}
	}
	
	var day: Int = 0
	var name: String = ""
	var emptyLinesIndicateMultipleInputs: Bool = true
	var visualizationEnabled = true
	
	@discardableResult func solve(_ input: AoCInput) -> AoCResult {
		print("Day \(String(format: "%02d", arguments: [day])): \(name) input: \(input.fileName) [\(input.index)]");
		return AoCResult(part1: "", part2: "")
	}
}
